swift-nav/libswiftnav
# Open source publish

Open source publish of df25faf9f28eb74ace9f9765c451087320b2d8ba..cae2886bb7ef4baa3c44864b5b637ea3d227a25f

Jira: https://swift-nav.atlassian.net/browse/DIP-307

- Install the leap seconds header
- updated leap second
- bazel: add swift_cc* wrappers
- bazel: fix code-coverage
- Remove bazel submodule
  - Fetches swift's bazel rules via http_archive instead of using a submodule
- Add cn0 mask to calc-pvt
- bazel: Require build passes in CI
- Export decode_sisa_index(), decode_bds_ura_index() and decode_fit_interval() functions: This allows external code to reuse libswiftnav's functions for decoding URA/SISA index values.
  - Also includes a couple of minor improvements:
    - Move definition of GAL_WEEK_TO_GPS_WEEK from ephemeris.h to gnss_time.h to be consistent with BDS_WEEK_TO_GPS_WEEK
    - Fix test for platforms where time_t is a signed 32-bit integer
- Added Makefile with bazel wrappers
- Fix group delay type for B1C
- bazel: fix optimized compilation
- clean up iono handling and navigation_measurement_t
- bazel add gen_compile_commands target
- bazel: make max_channels a configurable attribute
- bazel: use c only macros to build libswiftnav
- Handle endianess in bytestreams
- Byte swap inline function
- Regenerate leap second
- bazel: Add bazeliskrc file
- Use float whenever possible in satellite attitude and std
- Revert "Use float whenever possible in satellite attitude and std"
- Use float is satellite vel and acc when possible: Changed satellite acceleration and velocity to float when possible
- Handle BDS B3I in get_tgd_correction()
- Revert "Use float is satellite vel and acc when possible"
- Add support for GPS L1CQ in health check

---

Co-authored-by: Jason Mobarak <jason@swift-nav.com>
Co-authored-by: Ivan Smolyakov <ivan.smolyakov@swift-nav.com>
Co-authored-by: Altti Jokinen <altti.jokinen@swift-nav.com>
Co-authored-by: Marco <marco.mendonca@swift-nav.com>
Co-authored-by: Ivan Smolyakov <ivan.smolyakov@swift-nav.com>
Co-authored-by: Antriksh Srivastava <antriksh.srivastava@swift-nav.com>
Co-authored-by: Joseph Angelo <joseph.angelo@swift-nav.com>
Co-authored-by: Isaac Torres <isaac.torres@swift-nav.com>
Co-authored-by: Lucas Le <97814268+lucasle-sn@users.noreply.github.com>
Co-authored-by: Rodrigo Reichert <rodrigo.reichert@swift-nav.com>
Co-authored-by: Matt Woodward <46688854+woodfell@users.noreply.github.com>
Co-authored-by: Wiktor Więcław <110531204+wwieclaw@users.noreply.github.com>
Co-authored-by: Krzysztof Naglik <krzysztofnaglik96@gmail.com>
Co-authored-by: dgburr <dburr@dburr.net>
